File: swift/flatten-array/Sources/FlattenArray/FlattenArray.swift
func flattenArray(_ array:[Any]) -> [Int] {
    var out : [Int] = []
    func flatten(_ array:[Any]) {
        for i in array {
            if let a = i as? [Any] { flatten(a) } else { out.append(Int(i)) }
        }
    }
    flatten(array)
    return out
}

File: ruby/flatten-array/flatten_array.rb
class FlattenArray
  def self.flatten(unflat)
    out = []
    flattenize = ->(e) {
      e.kind_of?(Array) ? e.each { |e| flattenize.call(e) } : out.append(e) unless e.nil?
    }
    flattenize.call(unflat)
    return out
  end
end
